custom/abdm/models.py

```diff
@@ -0,0 +1,35 @@
+from django.db import models
+from datetime import datetime, timedelta
+from rest_framework.authtoken.models import Token
+from django.conf import settings
+
+
+class ABDMUser(models.Model):
+    username = models.CharField(max_length=100)
+    access_token = models.CharField(max_length=2000, null=True, blank=True)
+    token_created_at = models.DateTimeField()
+    domain = models.CharField(max_length=100)
+
+    class Meta:
+        unique_together = ['username', 'domain']
+
+    def save(self, *args, **kwargs):
+        if not self.access_token:
```

Comment:
Should this also consider checking if the token is expired?

Reply:
Instead of refreshing the token via user.save consider an explicit user.refresh_token (which refreshes and saves the object) and use that instead.

Reply:
Resolved in 69053d9798100104dee0963f127cabf48fdf0c98